types/index.esm.d.ts

```diff
@@ -1430,17 +1430,17 @@ export interface CoreChartOptions<TType extends ChartType> extends ParsingOption
    * base color
    * @see Defaults.color
    */
-  color: Scriptable<Color, ScriptableContext<TType>>;
+  color: ScriptableAndArray<Color, ScriptableContext<TType>>;
```

Comment:
This is not totally correct, for some elements like the ticks it does apply the array format correctly but the legend's color becomes black instead of any of the specified or default color. 

So either the legends behaviour needs a fix or I think that its better to keep it the way it is now since it does not support "bugged" behaviour.

https://jsfiddle.net/Leelenaleee/2dkL39ec/2/